src/utils/index.js

```diff
@@ -162,3 +171,22 @@ export const handleDBErrors = (error, { req, Sequelize }, cb) => {
   }
   return cb(errorResponseMessage);
 };
+
+/**
+ * @function addToReadHistory
+ * @param {string} articleId id of the article to be added
+ * @param {string} userId id of the user read stats to be added
+ * @returns {ArticleReadHistory} returns a the newly created article read history
+ */
+export const addArticleToReadHistory = async (articleId, userId) => {
+  try {
+    let articleReadHistory = await ArticleReadHistory.findOne({ where: { articleId, userId } });
+    if (!articleReadHistory) {
+      articleReadHistory = await ArticleReadHistory.create({ articleId, userId });
+      return articleReadHistory;
+    }
+    return articleReadHistory;
+  } catch (error) {
+    throw error;
```

Comment:
@micah-akpan Please, could you handle the error, given that it may differ in types. Nice Work

Reply:
@chikeozulumba Thanks for pointing that out. I considered doing that but deferred the error handling to where the function is used, since it is used in a function that has a try/catch, so if the error is thrown, it will be caught at that level. I will consider handling it here since this function might be used in other use cases.